src/uu/cp/src/copydir.rs

```diff
@@ -8,7 +8,7 @@
 //! See the [`copy_directory`] function for more information.
 #[cfg(windows)]
 use std::borrow::Cow;
-use std::collections::HashSet;
+use std::collections::{HashMap, HashSet};
```

Comment:
Is it possible to use a single `HashMap` for both symlinks and hardlinks or do they track different things?

Reply:
yes it can, but I think modifying this 
```
symlinked_files: &mut HashSet<FileInformation>,
```
is way beyond this PR's scope.

if you still feel ok, then I can do this modification.

Reply:
and yes I think they are tracking different things,
so we should not mess them together right now.

Reply:
Alright! Let's not merge them then.